File: src/main/java/com/github/unchama/seichiassist/data/subhome/SubHome.java
package com.github.unchama.seichiassist.data.subhome;

import org.bukkit.Location;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * サブホームオブジェクトのクラス
 */
public class SubHome {
    private final @NotNull Location location;
    public final @Nullable String name;

    public SubHome(@NotNull Location location, @Nullable String name) {
        this.location = location;
        this.name = name;
    }

    public Location getLocation() {
        // BukkitのLocationはミュータブルなのでコピーして返す必要がある
        return location.clone();
    }
}

File: src/main/scala/com/github/unchama/seichiassist/data/subhome/SubHome.scala
package com.github.unchama.seichiassist.data.subhome

import org.bukkit.Location

case class SubHome(location: Location, name: String) {
  def getLocation: Location = location.clone // BukkitのLocationはミュータブルなのでコピーして返す必要がある
}
